### DISCUSSION TITLE
add support for `pet search <search term>`

### DISCUSSION THREAD

**Discussion by sming** [knqyf263/pet#248] [Ideas]
#### I want to quickly see/execute a command by searching for a word that I know is unique to exactly one command/description

- I am a engineer
- Who wants to see/exec a command quickly, without intermediate questions/lists
- So that I can see/exec the command much faster and be more efficient overall

#### Proposed Solution

- modify`pet search` to also allow `pet search <search term>`
- add flag `--exec` to execute the match
- add flag `--list` to show the match
- if > 1 match, revert to default `pet search` behaviour or even better, pre-filter the results to those matching the search term
- add `[General]→defaultsearchflag` config option
    - options are `exec` or `list`

---

PS sorry if this is a duplicate. I looked and couldn't find one like this.

**Comment by RamiAwar** [comment]
Love this idea, seen several requests for it, will add to high prio feature list

**Comment by RamiAwar** [comment]
I think the real goal behind it is efficiency, will try to think of other approaches that speed things up and achieve this goal to see if there are any
